src/scheduler/server_info.cpp

```diff
@@ -431,6 +423,7 @@ query_server(status *pol, int pbs_sd)
 		return NULL;
 	}
 
+	/* Clean up some node tasks which we couldn't do in query_nodes() because we didn't have everything we needed */
```

Comment:
I am a little confused here, what are we trying to clean here?

Reply:
I'll update the comment.  I was just trying to be terse.  I was a little too terse.  Ideally we'd do everything to query the nodes in query_nodes().  Unfortunately we don't have all the data we need to do that.  Some of it is queried/created after the query_nodes() call.  It had to be delayed until near the end of query_server().